doc/source/workflow.rst

```diff
@@ -0,0 +1,726 @@
+   that will occur during the ``create`` step, that are difficult to track. The
+   recommended way to apply changes to the unpacked image directory is to change
+   the configuration and re-run the ``prepare`` step.
+
+
+The Create Step
+...............
+
+The successful completion of the ``prepare`` step is a prerequisite for the
+``create`` step. It ensures the unpacked root tree is complete and consistent.
+Creating the packed, or final, image is done in the ``create`` step. Multiple
+images can be created using the same unpacked root tree. It is, for example,
+possible to create a self installing OEM image and a virtual machine image from
+a single unpacked root tree. The only prerequisite is that both image types are
+specified in the :file:`config.xml` before the prepare step is executed.
+
+During the ``create`` step the following major operations are performed by
+kiwi:
+
+#. **Execute the User-defined Script** ``images.sh``.
+
+   At the beginning of the image creation process the script named
+   :file:`images.sh` is executed if present. It is executed on the top level of
+   the target root tree. The script is usually used to remove files that are no
+   needed in the final image. For example, if an appliance is being built for a
+   specific hardware, unnecessary kernel drivers can be removed using this
+   script. 
+   
+#. **Create Requested Image Type.** 
+
+   The image types that can be created from a prepared image tree depend on the
+   types specified in the image description file:`config.xml` file. The
+   configuration file must contain at least one ``type`` element. 
+  
+   The currently supported image types are:
+   
+   #. **Live Image**: For CDs,DVDs or flash disks.
+
+   #. **Disk image**: Virtual system disk that can be used in virtual
+      environments such as VMware, Xen, Amazon Cloud, KVM, and others. Depending
+      on the format a guest configuration file is created. 
+
+   #. **OEM Image**: Preload system for install media CD/DVD or flash disk.
+
+   #. **PXE Image**: Network boot image.
+
+   #. **Container Image**: Base images for the Docker engine.
+
+Customizing the Boot Process
+----------------------------
+
+Most Linux systems use a special boot image to control the system boot process
+after the system firmware, BIOS or UEFI, hands control of the hardware to the
+operating system. This boot image is called the initrd. The Linux kernel loads
+the *initrd*, a compressed cpio initial RAM disk, into the RAM and executes
+*init* or, if present, *linuxrc*.
+
+Depending on the image type, KIWI creates the boot image automatically during
+the ``create`` step. Each image type has its own description for the boot image.
+Common functionality is shared between the boot images through a set of
+functions. The boot image descriptions follow the same principles as the system
+image descriptions, KIWI ships with pre-defined boot image descriptions.
+
+.. topic:: Note: Boot Image Descriptions provided by KIWI
```

Comment:
It would be great if we can add a short information that this version of kiwi supports both:

* kiwi generic boot image using kiwi boot code and supporting all the nice modes e.g oem install, pxe install, netclients, live image, and so on
* dracut generated boot image useful for fast boot, small initrd images, but only with limited features and all the variable documentation here applies only to the kiwi boot code